flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/ZooKeeperCheckpointIDCounterITCase.java

```diff
@@ -36,16 +37,21 @@
  */
 public class ZooKeeperCheckpointIDCounterITCase extends CheckpointIDCounterTestBase {
 
-    private static final ZooKeeperTestEnvironment ZooKeeper = new ZooKeeperTestEnvironment(1);
+    private static ZooKeeperTestEnvironment zookeeper;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
```

Comment:
> [hotfix][tests] Allow ZooKeeperCheckpointIDCounterITCase to loop

Out of curiosity: Is this required to run the Intellij run-until-failure functionality? 🤔  I couldn't reproduce it. In what sense is the new code different from the old version?

Reply:
it depends how you set up intellij. If it initializes the class for each test run it of course works as is, but if that's not the case then there is no matching `@BeforeClass` to reset zookeeper.

Reply:
But if for example we'd add a junit retry to the test it would also start failing afaik.

Reply:
> But if for example we'd add a junit retry to the test it would also start failing afaik.
Makes sense. I tried it out of curiosity: Interestingly, reverting the `@BeforeAll` annotated `setUp` method and doing the instantiation in the member variable declaration still works even when using `@RepeatedTest` on one of the tests still works 🤔 

Anyway, it's looks more consistent 👍